### PR TITLE
Add omp impl for tryparallelfor and modify gelu to use fastgelu impl.

### PR DESCRIPTION
**Pull request by pranavsharma**
**Description**: Add omp impl for tryparallelfor and modify gelu to use fastgelu impl.

**Motivation and Context**
@tianleiwu reported recently that the Bert model regressed in perf after changes in PR 3153.

### CODE REVIEW

File: include/onnxruntime/core/platform/threadpool.h [hunk 1]
```diff
@@ -164,6 +164,22 @@ class ThreadPool {
   void ParallelFor(std::ptrdiff_t total, const TensorOpCost& cost_per_unit,
                    const std::function<void(std::ptrdiff_t first, std::ptrdiff_t)>& fn);
 
+#ifdef _OPENMP
+  static void TryParallelFor(concurrency::ThreadPool* tp, std::ptrdiff_t total, const TensorOpCost&,
+                             const std::function<void(std::ptrdiff_t first, std::ptrdiff_t last)>& fn) {
+    std::ptrdiff_t num_threads = concurrency::ThreadPool::NumThreads(tp);
```
Comment (skottmckay) [inline include/onnxruntime/core/platform/threadpool.h:170]: what's the reason for including the function signature within the ifdef vs using ORT_UNUSED_PARAMETER for cost_per_unit if the #ifdef was just for the whole body?

File: include/onnxruntime/core/platform/threadpool.h [hunk 2]
```diff
@@ -172,13 +188,30 @@ class ThreadPool {
...
   // Similar to ParallelFor above, but takes the specified scheduling strategy
   // into account.
   void
   ParallelFor(std::ptrdiff_t total, const SchedulingParams& scheduling_params,
               const std::function<void(std::ptrdiff_t, std::ptrdiff_t)>& fn);
 
+#ifdef _OPENMP
+  static void TryParallelFor(concurrency::ThreadPool* tp, std::ptrdiff_t total, const SchedulingParams&,
+                             const std::function<void(std::ptrdiff_t, std::ptrdiff_t)>& fn) {
+    std::ptrdiff_t num_threads = concurrency::ThreadPool::NumThreads(tp);
```
Comment (skottmckay) [inline include/onnxruntime/core/platform/threadpool.h:202]: nit: slightly more readable if there are names for the fn params like

                   const std::function<void(std::ptrdiff_t first, std::ptrdiff_t last)>& fn);

File: include/onnxruntime/core/platform/threadpool.h (hunk 1 above)
Comment (pranavsharma) [reply include/onnxruntime/core/platform/threadpool.h:170]: Not sure. I was following what TryBatchParallelFor did :-). Let me see if using ORT_UNUSED_PARAMETER works.

File: onnxruntime/contrib_ops/cpu/activations.h [hunk 3]
```diff
@@ -37,23 +37,36 @@ class Gelu : public OpKernel {
...
+        tp, static_cast<int32_t>(task_count),
+        [&](ptrdiff_t task_idx) {
+          const auto start = task_idx * length_per_task;
+          const T* p_input = input_data + start;
+          T* p_output = output_data + start;
+          int64_t count = std::min(length_per_task, elem_count - start);
+
+          for (int64_t i = 0; i < count; i++) {
+            T value = p_input[i];
+            p_output[i] = value * static_cast<float>(M_SQRT1_2);
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/activations.h:60]: static_cast<T>(M_SQRT1_2)

**Pull request merged**